### PR TITLE
Add merchants resource

### PR DESCRIPTION
**Pull request by griffinmyers**
This PR adds support for the merchants resource. 

A followup PR will cut a minor version release.

### CODE REVIEW

File: lib/resources/merchants.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+'use strict';
+
+var merge = require('../merge');
+var formatQuery = require('../formatQuery');
+
+function merchants(requestOptions, maybePromiseRequest) {
+  return {
+    all: function all(params, callback) {
+      //
+      // Gets an index of all merchants.
```
Comment (gckwan) [inline lib/resources/merchants.js]: I'd suggest listing the options under @params too, so someone reading just this file would know that filtering options are available.

File: README.md [hunk 2]
```diff
@@ -118,12 +118,88 @@ The `data` key will contain any resource data received from the API and the `met
...
+});
+```
+
+### Merchants
+
+##### All
+
+###### options
+
+* `status`: Status to filter by. One of ('approved', 'pending', or 'available')
```
Comment (gckwan) [inline README.md]: s/Status/Partnership request status, so it's clear what status we're filtering by :)

File: lib/resources/merchants.js (hunk 1 above)
Comment (griffinmyers) [inline lib/resources/merchants.js]: 👍 ! missed this

**Comment by griffinmyers** [comment]
thanks @gckwan !

**Pull request merged**